.github/workflows/check.yml

```diff
@@ -1,81 +1,229 @@
-      # in the PR. If they were, run the relevant linters. Otherwise, skip.
+      - name: Cloning repository
+        uses: actions/checkout@v3
 
-      - name: Lint Markdown
-        if: ${{ steps.filter.outputs.md == 'true' }}
-        run: npm run lint:md
+      - name: Setting up Node.js
+        uses: actions/setup-node@v3
+        with:
+          cache: npm
+          node-version: 16
 
-      - name: Lint JavaScript
-        if: ${{ steps.filter.outputs.js == 'true' || steps.filter.outputs.json == 'true' }}
-        run: npm run lint:js && npm run lint:types
+      - name: Installing Node.js packages
+        run: npm ci
 
-      - name: Build JavaScript
-        if: ${{ steps.filter.outputs.js == 'true' || steps.filter.outputs.json == 'true' }}
-        run: npm run rollup
+      - name: Linting JavaScript
+        if: ${{ steps.changes.outputs.js == 'true' }}
+        run: |
+          npm run lint:js
+          npm run lint:types
 
-      - name: Lint SCSS
-        if: ${{ steps.filter.outputs.scss == 'true' }}
+      - name: Linting Markdown
+        if: ${{ steps.changes.outputs.md == 'true' }}
+        run: npm run lint:md
+
+      - name: Linting SCSS
+        if: ${{ steps.changes.outputs.scss == 'true' }}
         run: npm run lint:scss
 
-      - name: Build External
-        run: npm run build-external
+  test:
+    needs: queue
+    runs-on: ubuntu-latest
+    steps:
+      - name: Getting changes
+        id: changes
+        uses: dorny/paths-filter@v2
+        with:
+          filters: |
+            js:
+              - 'package.json'
+              - 'site/**/*.js'
+              - 'server/**.js'
+              - 'tests/**.js'
+
+      - name: Cloning repository
+        uses: actions/checkout@v3
+
+      - name: Setting up Node.js
+        uses: actions/setup-node@v3
+        with:
+          cache: npm
+          node-version: 16
+
+      - name: Installing Node.js packages
+        run: npm ci
+
+      - name: Running tests
+        if: ${{ steps.changes.outputs.js == 'true' }}
+        run: |
+          npm run test
+
+  build:
+    needs: queue
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cloning repository
+        uses: actions/checkout@v3
+
+      - name: Setting up Node.js
+        uses: actions/setup-node@v3
+        with:
+          cache: npm
+          node-version: 16
+
+      - name: Installing Node.js packages
+        run: npm ci
+
+      - name: Building site
+        env:
+          # Increase memory limit as a full build requires around 4GB
+          NODE_OPTIONS: --max_old_space_size=4096
+          ELEVENTY_IGNORE_HANDBOOK: true
+          ELEVENTY_IGNORE_LIGHTHOUSE: true
+          ELEVENTY_IGNORE_MULTIDEVICE: true
+          ELEVENTY_IGNORE_NACL: true
+          ELEVENTY_IGNORE_PRIVACY_SANDBOX: true
+          ELEVENTY_IGNORE_VERSIONHISTORY: true
+          ELEVENTY_IGNORE_WEBSTORE: true
+          ELEVENTY_IGNORE_WORKBOX: true
+          # ELEVENTY_INCLUDE_TRANSLATED: true
```

Comment:
This can be trimmed down once https://github.com/GoogleChrome/developer.chrome.com/pull/4316 is in.